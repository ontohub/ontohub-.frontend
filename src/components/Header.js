import React from 'react'
import { Block } from 'jsxstyle'
import { colors, sizes } from '../styles'

const Header = (props) => (
  <Block
    backgroundColor={colors.dark}
    color={`${colors.light} !important`}
    marginBottom="1em"
  >
    <Block maxWidth={sizes.contentWidth} margin="auto">
      {props.children}
    </Block>
  </Block>
)

export default Header
