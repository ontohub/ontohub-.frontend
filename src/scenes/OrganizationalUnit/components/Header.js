import React, { Fragment } from "react";
import { Header as SemHeader, Container, Menu, Icon } from "semantic-ui-react";
import { NavLink as Link } from "react-router-dom";
import Gravatar from "react-gravatar";

export const Header = ({
  data: {
    loading,
    organizationalUnit: {
      __typename,
      description,
      displayName,
      emailHash,
      id,
      permissions: { role } = {}
    } = {}
  }
}) => (
  <Container>
    <SemHeader as="h2" inverted>
      {!loading && (
        <Fragment>
          {emailHash && (
            <Gravatar
              key="avatar"
              style={{ borderRadius: 2 }}
              size={100}
              md5={emailHash}
            />
          )}
          <SemHeader.Content key="content">
            {displayName ? (
              <Fragment>
                {displayName}
                <SemHeader.Subheader key="id">
                  {description || id}
                </SemHeader.Subheader>
              </Fragment>
            ) : (
              id
            )}
          </SemHeader.Content>
        </Fragment>
      )}
    </SemHeader>
    <Menu inverted pointing secondary>
      <Menu.Item as={Link} to={`/${id}/repositories`}>
        <Icon name="fork" />
        Repositories
      </Menu.Item>
      {__typename === "User" ? (
        <Menu.Item as={Link} to={`/${id}/organizations`}>
          <Icon name="users" />
          Organizations
        </Menu.Item>
      ) : (
        <Menu.Item as={Link} to={`/${id}/members`}>
          <Icon name="users" /> Members
        </Menu.Item>
      )}
      {role === "admin" && (
        <Menu.Menu position="right">
          <Menu.Item as={Link} to={`/${id}/settings`}>
            <Icon name="settings" /> Settings
          </Menu.Item>
        </Menu.Menu>
      )}
    </Menu>
  </Container>
);
