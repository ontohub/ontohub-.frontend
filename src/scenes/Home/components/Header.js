import React from "react";
import { Header as SemHeader, List } from "semantic-ui-react";
import { PaddedContainer } from "../../../components";
import { Item } from ".";

export const Header = () => (
  <PaddedContainer>
    <SemHeader as="h2" inverted>
      Welcome to Ontohub
    </SemHeader>
    <List size="large">
      <Item icon="fork" header="Open" content="based on open source software" />
      <Item
        icon="code"
        header="Flexible"
        content="supporting OWL, UML, FOL/TPTP, HOL/THF, and more"
      />
      <Item
        icon="sitemap"
        header="Distributed"
        content="OMS alignments, mappings, networks, combinations using DOL"
      />
    </List>
  </PaddedContainer>
);

export default Header;
