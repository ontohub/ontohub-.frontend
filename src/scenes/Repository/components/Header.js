import React from "react";
import { get } from "lodash";
import { Link } from "react-router-dom";
import { PaddedContainer } from "../../../components";
import { Header as SemHeader } from "semantic-ui-react";

const PureHeader = ({
  match: { params: { organizationalUnitId, repositoryId } },
  ...props
}) => {
  console.log(props);
  return (
    <PaddedContainer>
      <SemHeader inverted as="h2">
        <Link to={`/${organizationalUnitId}`}>{organizationalUnitId}</Link> /{" "}
        {repositoryId}
        <SemHeader.Subheader key="description">
          {get(props, "data.repository.description")}
        </SemHeader.Subheader>
      </SemHeader>
    </PaddedContainer>
  );
};

export { PureHeader as Header };
