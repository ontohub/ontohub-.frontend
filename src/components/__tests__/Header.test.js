import React from 'react'
import { Header, PaddedContainer } from '../Header'
import toJSON from 'enzyme-to-json'
import { ThemeProvider } from 'styled-components'
import { render } from 'enzyme'

describe('Header', () => {
  it('matches the snapshot', () => {
    const wrapper = render(
      <ThemeProvider
        theme={{ colors: { dark: '#000' }, sizes: { contentWidth: '700px' } }}
      >
        <Header>
          Header content
        </Header>
      </ThemeProvider>
    )
    expect(toJSON(wrapper)).toMatchSnapshot()
  })
})

describe('PaddedContainer', () => {
  it('matches the snapshot', () => {
    const wrapper = render(
      <ThemeProvider theme={{ sizes: { headerPadding: '1em' } }}>
        <PaddedContainer>
          Header content
        </PaddedContainer>
      </ThemeProvider>
    )
    expect(toJSON(wrapper)).toMatchSnapshot()
  })
})
