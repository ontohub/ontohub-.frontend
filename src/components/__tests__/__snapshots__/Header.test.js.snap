// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches the snapshot 1`] = `
<div
  class="sc-bdVaJa gtWvUm"
>
  <div
    class="sc-bwzfXH cEnUnG"
  >
    Header content
  </div>
</div>
`;

exports[`PaddedContainer matches the snapshot 1`] = `
<div
  class="ui container sc-htpNat gTtniJ"
>
  Header content
</div>
`;
